api/v1beta2/network_types.go

```diff
@@ -446,6 +475,45 @@ func (s *SubnetSpec) String() string {
 	return fmt.Sprintf("id=%s/az=%s/public=%v", s.GetResourceID(), s.AvailabilityZone, s.IsPublic)
 }
 
+// IsEdge returns the true when the subnet is created in the edge zone,
+// Local Zones.
+func (s *SubnetSpec) IsEdge() bool {
+	if s.ZoneType == nil {
+		return false
+	}
+	if s.ZoneType.Equal(ZoneType(ZoneTypeLocalZone)) {
+		return true
+	}
+	return false
+}
+
+// SetZoneInfo updates the subnets with zone information.
+func (s *SubnetSpec) SetZoneInfo(zones []*ec2.AvailabilityZone) error {
+	zoneInfo := func(zoneName string) *ec2.AvailabilityZone {
+		for _, zone := range zones {
+			if aws.StringValue(zone.ZoneName) == zoneName {
+				return zone
+			}
+		}
+		return nil
+	}
+
+	zone := zoneInfo(s.AvailabilityZone)
+	if zone == nil {
+		if len(s.AvailabilityZone) > 0 {
+			return fmt.Errorf("unable to update zone information for subnet '%v' and zone '%v'", s.ID, s.AvailabilityZone)
+		}
+		return fmt.Errorf("unable to update zone information for subnet '%v'", s.ID)
+	}
+	if zone.ZoneType != nil {
+		s.ZoneType = ptr.To(ZoneType(*zone.ZoneType))
```

Comment:
Maybe I'm just being paranoid, but I feel like it's safer to save the value to a local var before taking its address:
```suggestion
        var zt := ZoneType(*zone.ZoneType)
		s.ZoneType = ptr.To(zt)
```

Reply:
Oh, I didn't mean safer in the sense of the value being `nil` but more in the sense of saving the addr of a temporary variable. I guess that doesn't make sense since it's basically doing `&(*zonetype)` which is `zonetype`.